hdkeychain/extendedkey.go

```diff
@@ -295,9 +297,9 @@ func (k *ExtendedKey) Child(i uint32) (*ExtendedKey, error) {
 		// derive the final child key.
 		//
 		// childKey = serP(point(parse256(Il)) + parentKey)
-		childX, childY := chainec.Secp256k1.Add(ilx, ily, pubKey.GetX(),
+		childX, childY := curve.Add(ilx, ily, pubKey.GetX(),
```

Comment:
Why call `.GetX`?  `X` is available on the struct directly.  In fact, `GetX` should ultimately be removed entirely from the `secp256k1.PublicKey` (it can't yet since it's still required for the `chainec` interface) since it doesn't even follow effective Go guidelines about getters.